traits/src/currency.rs

```diff
@@ -212,7 +213,12 @@ pub trait BasicCurrency<AccountId> {
 	// Public mutables
 
 	/// Transfer some amount from one account to another.
-	fn transfer(from: &AccountId, to: &AccountId, amount: Self::Balance) -> DispatchResult;
+	fn transfer(
+		from: &AccountId,
+		to: &AccountId,
+		amount: Self::Balance,
+		existence_requirement: ExistenceRequirement,
```

Comment:
this will break too many existing code.
I was thinking only update the transfer implementation of orml-tokens. Add a new helper method to take existence_requirement.